easybuild/tools/filetools.py

```diff
@@ -760,6 +760,17 @@ def verify_checksum(path, checksums):
         checksums = [checksums]
 
     for checksum in checksums:
+        if isinstance(checksum, dict):
+            filename = os.path.split(path)[-1]
+            if filename in checksum:
+                checksum = checksum[filename]
+            else:
+                # No checksum for this file
+                if build_option('enforce_checksums'):
+                    raise EasyBuildError("Missing checksum for %s", os.path.basename(path))
+                else:
+                    continue
+
         if isinstance(checksum, basestring):
```

Comment:
This should be an `elif` now?

I would also prefer having the more common case of a checksum as a string considered first, so:

```python
if isinstance(checksum, basestring):
    ...
elif isinstance(checksum, dict):
    ...